backend/src/services/trading_service.py

```
@@ -317,12 +317,16 @@ async def _confirm_transaction(self, signature: Signature, timeout: int = 60) ->
                     status = response.value[0]
                     if status.err:
-                        logger.error(f"Transaction failed: {status.err}")
+                        logger.error(f"Transaction failed for {signature}: {status.err}")
+                        # If it's a specific RPC error, log it for debugging
+                        if hasattr(status.err, '__dict__'):
+                            logger.error(f"Detailed RPC error: {status.err.__dict__}")
                         return False
                     if status.confirmation_status in ['confirmed', 'finalized']:
+                        logger.info(f"Transaction confirmed: {signature} ({status.confirmation_status})")
                         return True
             except Exception as e:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)